cartographer/io/in_memory_proto_stream.h

```diff
@@ -27,6 +27,25 @@
 namespace cartographer {
 namespace io {
 
+class ForwardingProtoStreamWriter
+    : public cartographer::io::ProtoStreamWriterInterface {
+ public:
+  // A callback that is invoked anytime 'WriteProto()' is called on the
+  // 'ForwardingProtoStreamWriter'. When 'Close()' is called on the
+  // 'ForwardingProtoStreamWriter' the callback is invoked with a 'nullptr'.
+  using WriterCallback =
+      std::function<bool(const google::protobuf::Message* proto)>;
+
+  explicit ForwardingProtoStreamWriter(WriterCallback writer_callback);
```

Comment:
Let's put constructor implementation to the header, like for InMemoryProtoStreamReader.

Reply:
Done.